flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/internal/FlinkOffsetBackingStore.java

```diff
@@ -0,0 +1,202 @@
+
+    @Override
+    public void configure(WorkerConfig config) {
+        // eagerly initialize the executor, because OffsetStorageWriter will use it later
+        start();
+
+        Map<String, ?> conf = config.originals();
+        if (!conf.containsKey(OFFSET_STATE_VALUE)) {
+            // a normal startup from clean state, not need to initialize the offset
+            return;
+        }
+
+        String stateJson = (String) conf.get(OFFSET_STATE_VALUE);
+        DebeziumOffsetSerializer serializer = new DebeziumOffsetSerializer();
+        DebeziumOffset debeziumOffset;
+        try {
+            debeziumOffset = serializer.deserialize(stateJson.getBytes(StandardCharsets.UTF_8));
+        } catch (IOException e) {
+            LOG.error("Can't deserialize debezium offset state from JSON: " + stateJson, e);
+            throw new RuntimeException(e);
+        }
+
+        String engineName = (String) conf.get(EmbeddedEngine.ENGINE_NAME.name());
+        Converter keyConverter = new JsonConverter();
+        Converter valueConverter = new JsonConverter();
+        keyConverter.configure(config.originals(), true);
+        Map<String, Object> valueConfigs = new HashMap<>(conf);
+        valueConfigs.put("schemas.enable", false);
+        valueConverter.configure(valueConfigs, true);
+        OffsetStorageWriter offsetWriter =
+                new OffsetStorageWriter(
+                        this,
+                        // must use engineName as namespace to align with Debezium Engine
+                        // implementation
+                        engineName,
+                        keyConverter,
+                        valueConverter);
+
+        offsetWriter.offset(debeziumOffset.sourcePartition, debeziumOffset.sourceOffset);
+
+        // flush immediately
+        if (!offsetWriter.beginFlush()) {
+            // if nothing is needed to be flushed, there must be something wrong with the
+            // initialization
+            LOG.warn(
+                    "Initialize FlinkOffsetBackingStore from empty offset state, this shouldn't happen.");
+            return;
+        }
+
+        // trigger flushing
+        Future<Void> flushFuture =
+                offsetWriter.doFlush(
+                        (error, result) -> {
+                            if (error != null) {
+                                LOG.error("Failed to flush initial offset.", error);
+                            } else {
+                                LOG.debug("Successfully flush initial offset.");
+                            }
+                        });
+
+        // wait until flushing finished
+        try {
+            flushFuture.get(FLUSH_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            LOG.info(
+                    "Flush offsets successfully, partition: {}, offsets: {}",
+                    debeziumOffset.sourcePartition,
+                    debeziumOffset.sourceOffset);
+        } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)